src/savegame_json.cpp

```diff
@@ -1886,6 +1889,10 @@ void monster::store( JsonOut &json ) const
     }
     json.member( "inv", inv );
 
+    if( dragged_foe ) {
+        json.member( "dragged_foe_id", dragged_foe->getID() );
```

Comment:
I haven't tested, but I suspect the cause of #33051 might be that the object dragged_foe points to has already been destructed here, leading to a use-after-free problem. It's probably better to always store dragged_foe_id in `monster` and retrieve the pointer on demand, which should eliminate the crash. (This is also how passenger saving/loading are done in the vehicle code, if I understood the code correctly).